internal/mode/static/state/dataplane/configuration_test.go

```diff
@@ -3019,9 +3047,74 @@ func TestBuildTelemetry(t *testing.T) {
 				},
 				NginxProxy: telemetryConfigured,
 			},
-			expTelemetry: expTelemetryConfigured,
+			expTelemetry: createTelemetry(),
 			msg:          "Telemetry configured",
 		},
+		{
+			g: &graph.Graph{
+				Gateway: &graph.Gateway{
+					Source: &v1.Gateway{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "gw",
+							Namespace: "ns",
+						},
+					},
+				},
+				NginxProxy: telemetryConfigured,
+				NGFPolicies: map[graph.PolicyKey]*graph.Policy{
+					{NsName: types.NamespacedName{Name: "obsPolicy"}}: {
+						Source: &ngfAPI.ObservabilityPolicy{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "obsPolicy",
+								Namespace: "custom-ns",
+							},
+							Spec: ngfAPI.ObservabilityPolicySpec{
+								Tracing: &ngfAPI.Tracing{
+									Ratio: helpers.GetPointer[int32](25),
+								},
+							},
+						},
+					},
+				},
+			},
+			expTelemetry: createModifiedTelemetry(func(t Telemetry) Telemetry {
+				t.Ratios = []Ratio{
+					{Name: "$otel_ratio_25", Value: 25},
+				}
+				return t
+			}),
+			msg: "Telemetry configured with observability policy ratio",
```

Comment:
Can we add a tests where there are several policies configured to make sure we are building the ratios correctly and ignoring non-observability policies?